models/report.js

```diff
@@ -63,6 +70,17 @@ schema.post('save', function() {
   }
 });
 
+schema.methods.flag = function() {
+  this.flagged = true;
+  this.markAsRead();
+};
+
+schema.methods.markAsRead = function() {
+  this.read = true;
+  this.checkedOutBy = null;
+  this.checkedOutAt = null;
```

Comment:
@hooverlunch these two functions are only used during the batch mode. I will send you an email with the explanation.